pkg/multiclustermanager/routes.go

```diff
@@ -111,9 +110,21 @@ func router(ctx context.Context, localClusterEnabled bool, tunnelAuthorizer *mcm
 	authed.PathPrefix("/v3/token").Handler(tokenAPI)
 	authed.PathPrefix("/v3").Handler(managementAPI)
 
+	// Authenticated routes
+	metricsAuthed := mux.NewRouter()
+	metricsAuthed.UseEncodedPath()
+	impersonsatingAuth := auth.ToMiddleware(requests.NewImpersonatingAuth(sar.NewSubjectAccessReview(clusterManager)))
+	tokenReviewAuth := impersonsatingAuth.Chain(auth.ToMiddleware(requests.NewTokenReviewAuth(scaledContext.K8sClient.AuthenticationV1())))
+	metricsAuthed.Use(mux.MiddlewareFunc(tokenReviewAuth))
+	metricsAuthed.Use(mux.MiddlewareFunc(rbac.NewAccessControlHandler()))
+	metricsAuthed.Use(requests.NewAuthenticatedFilter)
```

Comment:
I was hoping there would be an easier way to add a middleware to one route without this duplicate code but it looks like I was mistaken

Reply:
Thanks to some of Arvind's feedback I was able to reuse certain parts.